// @codex-data/sdk@1.0.3/dist/sdk/Mutation.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Mutation = void 0;
// THIS FILE IS AUTOMATICALLY GENERATED, DO NOT EDIT IT DIRECTLY
const graphql_1 = require("./generated/graphql");
class Mutation {
    constructor(sdk) {
        this.sdk = sdk;
        this.deleteApiToken = (vars) => __awaiter(this, void 0, void 0, function* () { return this.sdk.mutation(graphql_1.DeleteApiTokenDocument, vars); });
        this.deleteWebhooks = (vars) => __awaiter(this, void 0, void 0, function* () { return this.sdk.mutation(graphql_1.DeleteWebhooksDocument, vars); });
        this.createApiTokens = (vars) => __awaiter(this, void 0, void 0, function* () { return this.sdk.mutation(graphql_1.CreateApiTokensDocument, vars); });
        this.createWebhooks = (vars) => __awaiter(this, void 0, void 0, function* () { return this.sdk.mutation(graphql_1.CreateWebhooksDocument, vars); });
    }
}
exports.Mutation = Mutation;
