// graphql-ast-types@1.0.2/lib/definitions/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = defineType;
exports.chain = chain;
exports.assertEach = assertEach;
exports.assertOneOf = assertOneOf;
exports.assertNodeType = assertNodeType;
exports.assertNodeOrValueType = assertNodeOrValueType;
exports.assertValueType = assertValueType;
exports.assertArrayOf = assertArrayOf;
const t = require("../index");

const BUILDER_KEYS = exports.BUILDER_KEYS = {};
const NODE_FIELDS = exports.NODE_FIELDS = {};
const ALIAS_KEYS = exports.ALIAS_KEYS = {};

/**
 * Used to define an AST node.
 * @param {String} type The AST node name
 * @param {Object} opts Type definition object
 * @returns {void}
 */
function defineType(type, { fields = {}, aliases = [], builder = [] } = {}) {
  for (const key in fields) {
    const field = fields[key];

    // Sets field as optional if builder exist but validator does not.
    if (builder.indexOf(key) === -1) {
      field.optional = true;
    }
  }

  BUILDER_KEYS[type] = builder;
  NODE_FIELDS[type] = fields;
  ALIAS_KEYS[type] = aliases;
}

function getType(val) {
  if (Array.isArray(val)) {
    return "array";
  } else if (val === null) {
    return "null";
  } else if (val === undefined) {
    return "undefined";
  } else {
    return typeof val;
  }
}

// Validation helpers

function chain(...fns) {
  return function validate(...args) {
    fns.forEach(fn => fn(...args));
  };
}

function assertEach(callback) {
  function validator(node, key, val) {
    if (!Array.isArray(val)) {
      return;
    }

    val.forEach((it, i) => callback(node, `${key}[${i}]`, it));
  }
  return validator;
}

function assertOneOf(...vals) {
  function validate(node, key, val) {
    if (vals.indexOf(val.kind) < 0) {
      throw new TypeError(`Property ${key} expected value to be one of ${JSON.stringify(vals)} but got ${JSON.stringify(val)}`);
    }
  }

  return validate;
}

function assertNodeType(...types) {
  function validate(node, key, val) {
    const valid = types.every(type => t.is(type, val));

    if (!valid) {
      throw new TypeError(`Property ${key} of ${node.type} expected node to be of a type ${JSON.stringify(types)} ` + `but instead got ${JSON.stringify(val && val.type)}`);
    }
  }

  return validate;
}

function assertNodeOrValueType(...types) {
  function validate(node, key, val) {
    const valid = types.every(type => getType(val) === type || t.is(type, val));

    if (!valid) {
      throw new TypeError(`Property ${key} of ${node.type} expected node to be of a type ${JSON.stringify(types)} ` + `but instead got ${JSON.stringify(val && val.type)}`);
    }
  }

  return validate;
}

function assertValueType(type) {
  function validate(node, key, val) {
    const valid = getType(val) === type;

    if (!valid) {
      throw new TypeError(`Property ${key} expected type of ${type} but got ${getType(val)}`);
    }
  }

  return validate;
}

function assertArrayOf(cb) {
  return chain(assertValueType("array"), assertEach(cb));
}
