// @codex-data/sdk@1.0.3/dist/resources/graphql.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PoolNftType = exports.Plan = exports.ParallelAssetRankingAttribute = exports.ParallelAssetMatcherSubtype = exports.ParallelAssetMatcherRarity = exports.ParallelAssetMatcherParallel = exports.ParallelAssetMatcherClass = exports.ParallelAssetMatcherCardType = exports.PairRankingAttribute = exports.OrderType = exports.OrderState = exports.NftSearchable = exports.NftPoolType = exports.NftPoolRankingAttribute = exports.NftPoolEventType = exports.NftPoolContractVersion = exports.NftPoolCollectionRankingAttribute = exports.NftExchange = exports.NftEventTradeItemType = exports.NftEventOrderDirection = exports.NftContractErcType = exports.NftCollectionsLeaderboardMetric = exports.NftCollectionsLeaderboardDuration = exports.NftCollectionRankingAttribute = exports.NftAssetErrorStatus = exports.NftAssetAttributeType = exports.NftAssetAttributeDisplayType = exports.LiquidityProtocol = exports.LiquidityLockProtocol = exports.HoldersStatus = exports.HoldersSortAttribute = exports.GraphQlNftPoolVariant = exports.GasPrice = exports.ExchangeRankingAttribute = exports.EventType = exports.EventLabelType = exports.EventDisplayType = exports.DetailedStatsWindowSize = exports.DetailedPairStatsDuration = exports.DetailedNftStatsDuration = exports.CreationContext = exports.CostBasisMethod = exports.ContractType = exports.ContractLabelType = exports.ContractLabelSubType = exports.CommunityNoteType = exports.ChartTheme = exports.BondingCurveType = exports.AlertRecurrence = exports.AddressType = void 0;
exports.Link__Purpose = exports.Join__Graph = exports.WebhookType = exports.WebhookNftEventType = exports.WebhookNftEventFillSource = exports.TokenRankingAttribute = exports.TokenPairStatisticsType = exports.TokenPairEventType = exports.TokenOfInterest = exports.TokenLifecycleEventType = exports.SparklineAttribute = exports.SimulateTokenContractResultStatusEnum = exports.SimulateTokenContractBuySellErrorEnum = exports.SandwichLabelForEventType = exports.RankingDirection = exports.QuoteType = exports.QuoteToken = exports.QuoteCurrency = exports.PublishingType = exports.PrimePoolType = exports.PrimePoolEventType = exports.PrimePoolCurrency = void 0;
var AddressType;
(function (AddressType) {
    AddressType["Evm"] = "EVM";
    AddressType["Sol"] = "SOL";
    AddressType["Tron"] = "TRON";
})(AddressType || (exports.AddressType = AddressType = {}));
/** The recurrence of the webhook. Can be `INDEFINITE` or `ONCE`. */
var AlertRecurrence;
(function (AlertRecurrence) {
    AlertRecurrence["Indefinite"] = "INDEFINITE";
    AlertRecurrence["Once"] = "ONCE";
})(AlertRecurrence || (exports.AlertRecurrence = AlertRecurrence = {}));
/** The mathematical formula that defines how the prices of NFTs change after each buy or sell within a pool. */
var BondingCurveType;
(function (BondingCurveType) {
    BondingCurveType["Exponential"] = "EXPONENTIAL";
    BondingCurveType["Gda"] = "GDA";
    BondingCurveType["Linear"] = "LINEAR";
    BondingCurveType["Xyk"] = "XYK";
})(BondingCurveType || (exports.BondingCurveType = BondingCurveType = {}));
/** The color theme of the chart. */
var ChartTheme;
(function (ChartTheme) {
    ChartTheme["Dark"] = "DARK";
    ChartTheme["Light"] = "LIGHT";
})(ChartTheme || (exports.ChartTheme = ChartTheme = {}));
/** Type of the community gathered note. */
var CommunityNoteType;
(function (CommunityNoteType) {
    /** An contract attribute change. */
    CommunityNoteType["Attribute"] = "ATTRIBUTE";
    /** A logo change. */
    CommunityNoteType["Logo"] = "LOGO";
    /** A scam report. */
    CommunityNoteType["Scam"] = "SCAM";
})(CommunityNoteType || (exports.CommunityNoteType = CommunityNoteType = {}));
/** The contract label sub-type. */
var ContractLabelSubType;
(function (ContractLabelSubType) {
    ContractLabelSubType["Generic"] = "Generic";
    ContractLabelSubType["HighTax"] = "HighTax";
    ContractLabelSubType["HoneyPot"] = "HoneyPot";
    ContractLabelSubType["Imitator"] = "Imitator";
})(ContractLabelSubType || (exports.ContractLabelSubType = ContractLabelSubType = {}));
/** The contract label type. */
var ContractLabelType;
(function (ContractLabelType) {
    ContractLabelType["Scam"] = "Scam";
    ContractLabelType["Verified"] = "Verified";
})(ContractLabelType || (exports.ContractLabelType = ContractLabelType = {}));
var ContractType;
(function (ContractType) {
    ContractType["Nft"] = "NFT";
    ContractType["Token"] = "TOKEN";
})(ContractType || (exports.ContractType = ContractType = {}));
var CostBasisMethod;
(function (CostBasisMethod) {
    CostBasisMethod["Fifo"] = "FIFO";
})(CostBasisMethod || (exports.CostBasisMethod = CostBasisMethod = {}));
var CreationContext;
(function (CreationContext) {
    CreationContext["Mobile"] = "MOBILE";
    CreationContext["Telegram"] = "TELEGRAM";
    CreationContext["Web"] = "WEB";
})(CreationContext || (exports.CreationContext = CreationContext = {}));
/** The duration used to request detailed NFT stats. */
var DetailedNftStatsDuration;
(function (DetailedNftStatsDuration) {
    DetailedNftStatsDuration["Day1"] = "day1";
    DetailedNftStatsDuration["Day30"] = "day30";
    DetailedNftStatsDuration["Hour1"] = "hour1";
    DetailedNftStatsDuration["Hour4"] = "hour4";
    DetailedNftStatsDuration["Hour12"] = "hour12";
    DetailedNftStatsDuration["Week1"] = "week1";
})(DetailedNftStatsDuration || (exports.DetailedNftStatsDuration = DetailedNftStatsDuration = {}));
/** The duration used to request detailed pair stats. */
var DetailedPairStatsDuration;
(function (DetailedPairStatsDuration) {
    DetailedPairStatsDuration["Day1"] = "day1";
    DetailedPairStatsDuration["Day30"] = "day30";
    DetailedPairStatsDuration["Hour1"] = "hour1";
    DetailedPairStatsDuration["Hour4"] = "hour4";
    DetailedPairStatsDuration["Hour12"] = "hour12";
    DetailedPairStatsDuration["Min5"] = "min5";
    DetailedPairStatsDuration["Min15"] = "min15";
    DetailedPairStatsDuration["Week1"] = "week1";
})(DetailedPairStatsDuration || (exports.DetailedPairStatsDuration = DetailedPairStatsDuration = {}));
/** The window size used to request detailed stats. */
var DetailedStatsWindowSize;
(function (DetailedStatsWindowSize) {
    DetailedStatsWindowSize["Day1"] = "day1";
    DetailedStatsWindowSize["Hour1"] = "hour1";
    DetailedStatsWindowSize["Hour4"] = "hour4";
    DetailedStatsWindowSize["Hour12"] = "hour12";
    DetailedStatsWindowSize["Min5"] = "min5";
})(DetailedStatsWindowSize || (exports.DetailedStatsWindowSize = DetailedStatsWindowSize = {}));
/** A more specific breakdown of `EventType`. Splits `Swap` into `Buy` and `Sell`. */
var EventDisplayType;
(function (EventDisplayType) {
    EventDisplayType["Burn"] = "Burn";
    EventDisplayType["Buy"] = "Buy";
    EventDisplayType["Collect"] = "Collect";
    EventDisplayType["CollectProtocol"] = "CollectProtocol";
    EventDisplayType["Mint"] = "Mint";
    EventDisplayType["Sell"] = "Sell";
    EventDisplayType["Sync"] = "Sync";
})(EventDisplayType || (exports.EventDisplayType = EventDisplayType = {}));
/** The event label type. */
var EventLabelType;
(function (EventLabelType) {
    EventLabelType["FrontRun"] = "FrontRun";
    EventLabelType["Sandwiched"] = "Sandwiched";
})(EventLabelType || (exports.EventLabelType = EventLabelType = {}));
/** The event type for a token transaction. */
var EventType;
(function (EventType) {
    EventType["Burn"] = "Burn";
    EventType["Collect"] = "Collect";
    EventType["CollectProtocol"] = "CollectProtocol";
    EventType["Mint"] = "Mint";
    EventType["PoolBalanceChanged"] = "PoolBalanceChanged";
    EventType["Swap"] = "Swap";
    EventType["Sync"] = "Sync";
})(EventType || (exports.EventType = EventType = {}));
/** The attribute used to rank exchanges. */
var ExchangeRankingAttribute;
(function (ExchangeRankingAttribute) {
    ExchangeRankingAttribute["DailyActiveUsers"] = "dailyActiveUsers";
    ExchangeRankingAttribute["MonthlyActiveUsers"] = "monthlyActiveUsers";
    ExchangeRankingAttribute["TxnCount1"] = "txnCount1";
    ExchangeRankingAttribute["TxnCount4"] = "txnCount4";
    ExchangeRankingAttribute["TxnCount12"] = "txnCount12";
    ExchangeRankingAttribute["TxnCount24"] = "txnCount24";
    ExchangeRankingAttribute["VolumeNbt1"] = "volumeNBT1";
    ExchangeRankingAttribute["VolumeNbt4"] = "volumeNBT4";
    ExchangeRankingAttribute["VolumeNbt12"] = "volumeNBT12";
    ExchangeRankingAttribute["VolumeNbt24"] = "volumeNBT24";
    ExchangeRankingAttribute["VolumeUsd1"] = "volumeUSD1";
    ExchangeRankingAttribute["VolumeUsd4"] = "volumeUSD4";
    ExchangeRankingAttribute["VolumeUsd12"] = "volumeUSD12";
    ExchangeRankingAttribute["VolumeUsd24"] = "volumeUSD24";
})(ExchangeRankingAttribute || (exports.ExchangeRankingAttribute = ExchangeRankingAttribute = {}));
var GasPrice;
(function (GasPrice) {
    GasPrice["High"] = "HIGH";
    GasPrice["Low"] = "LOW";
    GasPrice["Medium"] = "MEDIUM";
})(GasPrice || (exports.GasPrice = GasPrice = {}));
/** The pool variant. */
var GraphQlNftPoolVariant;
(function (GraphQlNftPoolVariant) {
    GraphQlNftPoolVariant["Erc20"] = "ERC20";
    GraphQlNftPoolVariant["Native"] = "NATIVE";
})(GraphQlNftPoolVariant || (exports.GraphQlNftPoolVariant = GraphQlNftPoolVariant = {}));
var HoldersSortAttribute;
(function (HoldersSortAttribute) {
    HoldersSortAttribute["Balance"] = "BALANCE";
    HoldersSortAttribute["Date"] = "DATE";
})(HoldersSortAttribute || (exports.HoldersSortAttribute = HoldersSortAttribute = {}));
var HoldersStatus;
(function (HoldersStatus) {
    HoldersStatus["Disabled"] = "DISABLED";
    HoldersStatus["Enabled"] = "ENABLED";
})(HoldersStatus || (exports.HoldersStatus = HoldersStatus = {}));
/** Protocols that can lock liquidity */
var LiquidityLockProtocol;
(function (LiquidityLockProtocol) {
    LiquidityLockProtocol["BasecampV1"] = "BASECAMP_V1";
    LiquidityLockProtocol["Bitbond"] = "BITBOND";
    LiquidityLockProtocol["Burn"] = "BURN";
    LiquidityLockProtocol["UncxV2"] = "UNCX_V2";
    LiquidityLockProtocol["UncxV3"] = "UNCX_V3";
})(LiquidityLockProtocol || (exports.LiquidityLockProtocol = LiquidityLockProtocol = {}));
/** Protocols that create trading pairs */
var LiquidityProtocol;
(function (LiquidityProtocol) {
    LiquidityProtocol["RaydiumV4"] = "RAYDIUM_V4";
    LiquidityProtocol["UniswapV2"] = "UNISWAP_V2";
    LiquidityProtocol["UniswapV3"] = "UNISWAP_V3";
})(LiquidityProtocol || (exports.LiquidityProtocol = LiquidityProtocol = {}));
/** The display type for the NFT asset attribute. */
var NftAssetAttributeDisplayType;
(function (NftAssetAttributeDisplayType) {
    NftAssetAttributeDisplayType["BoostNumber"] = "BoostNumber";
    NftAssetAttributeDisplayType["BoostPercentage"] = "BoostPercentage";
    NftAssetAttributeDisplayType["Date"] = "Date";
    NftAssetAttributeDisplayType["Ranking"] = "Ranking";
    NftAssetAttributeDisplayType["Stat"] = "Stat";
    NftAssetAttributeDisplayType["Trait"] = "Trait";
})(NftAssetAttributeDisplayType || (exports.NftAssetAttributeDisplayType = NftAssetAttributeDisplayType = {}));
/** The type for the NFT asset attribute `value` field. */
var NftAssetAttributeType;
(function (NftAssetAttributeType) {
    NftAssetAttributeType["Array"] = "Array";
    NftAssetAttributeType["Number"] = "Number";
    NftAssetAttributeType["String"] = "String";
})(NftAssetAttributeType || (exports.NftAssetAttributeType = NftAssetAttributeType = {}));
var NftAssetErrorStatus;
(function (NftAssetErrorStatus) {
    NftAssetErrorStatus["IndexingInProgress"] = "INDEXING_IN_PROGRESS";
    NftAssetErrorStatus["NotFound"] = "NOT_FOUND";
})(NftAssetErrorStatus || (exports.NftAssetErrorStatus = NftAssetErrorStatus = {}));
/** The attribute used to rank NFT collections. */
var NftCollectionRankingAttribute;
(function (NftCollectionRankingAttribute) {
    NftCollectionRankingAttribute["LastEventTimestamp"] = "lastEventTimestamp";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenAverageChange"] = "stats1hNetworkBaseTokenAverageChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenAverageCurrent"] = "stats1hNetworkBaseTokenAverageCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenAveragePrevious"] = "stats1hNetworkBaseTokenAveragePrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenCloseChange"] = "stats1hNetworkBaseTokenCloseChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenCloseCurrent"] = "stats1hNetworkBaseTokenCloseCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenClosePrevious"] = "stats1hNetworkBaseTokenClosePrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenHighestSaleChange"] = "stats1hNetworkBaseTokenHighestSaleChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenHighestSaleCurrent"] = "stats1hNetworkBaseTokenHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenHighestSalePrevious"] = "stats1hNetworkBaseTokenHighestSalePrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenLowestSaleChange"] = "stats1hNetworkBaseTokenLowestSaleChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenLowestSaleCurrent"] = "stats1hNetworkBaseTokenLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenLowestSalePrevious"] = "stats1hNetworkBaseTokenLowestSalePrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenOpenChange"] = "stats1hNetworkBaseTokenOpenChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenOpenCurrent"] = "stats1hNetworkBaseTokenOpenCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenOpenPrevious"] = "stats1hNetworkBaseTokenOpenPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlendChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlendCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlendPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlurChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlurCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlurPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlurV2Change"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlurV2Current"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceBlurV2Previous"] = "stats1hNetworkBaseTokenVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceCryptopunksChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceGemChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceGemCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceGemPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceLooksrareChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change"] = "stats1hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current"] = "stats1hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous"] = "stats1hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceOpenseaChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceOpenseaProChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSeaportChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSeaportCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSeaportPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSudoswapChange"] = "stats1hNetworkBaseTokenVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent"] = "stats1hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious"] = "stats1hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change"] = "stats1hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current"] = "stats1hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous"] = "stats1hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceX2Y2Change"] = "stats1hNetworkBaseTokenVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceX2Y2Current"] = "stats1hNetworkBaseTokenVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeByFillsourceX2Y2Previous"] = "stats1hNetworkBaseTokenVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeChange"] = "stats1hNetworkBaseTokenVolumeChange";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumeCurrent"] = "stats1hNetworkBaseTokenVolumeCurrent";
    NftCollectionRankingAttribute["Stats1hNetworkBaseTokenVolumePrevious"] = "stats1hNetworkBaseTokenVolumePrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyMintsChange"] = "stats1hNonCurrencyMintsChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyMintsCurrent"] = "stats1hNonCurrencyMintsCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyMintsPrevious"] = "stats1hNonCurrencyMintsPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencySalesChange"] = "stats1hNonCurrencySalesChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencySalesCurrent"] = "stats1hNonCurrencySalesCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencySalesPrevious"] = "stats1hNonCurrencySalesPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyTokensSoldChange"] = "stats1hNonCurrencyTokensSoldChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyTokensSoldCurrent"] = "stats1hNonCurrencyTokensSoldCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyTokensSoldPrevious"] = "stats1hNonCurrencyTokensSoldPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyTransfersChange"] = "stats1hNonCurrencyTransfersChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyTransfersCurrent"] = "stats1hNonCurrencyTransfersCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyTransfersPrevious"] = "stats1hNonCurrencyTransfersPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueBuyersChange"] = "stats1hNonCurrencyUniqueBuyersChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueBuyersCurrent"] = "stats1hNonCurrencyUniqueBuyersCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueBuyersPrevious"] = "stats1hNonCurrencyUniqueBuyersPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueMintersChange"] = "stats1hNonCurrencyUniqueMintersChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueMintersCurrent"] = "stats1hNonCurrencyUniqueMintersCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueMintersPrevious"] = "stats1hNonCurrencyUniqueMintersPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueSalesWalletsChange"] = "stats1hNonCurrencyUniqueSalesWalletsChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueSalesWalletsCurrent"] = "stats1hNonCurrencyUniqueSalesWalletsCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueSalesWalletsPrevious"] = "stats1hNonCurrencyUniqueSalesWalletsPrevious";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueSellersChange"] = "stats1hNonCurrencyUniqueSellersChange";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueSellersCurrent"] = "stats1hNonCurrencyUniqueSellersCurrent";
    NftCollectionRankingAttribute["Stats1hNonCurrencyUniqueSellersPrevious"] = "stats1hNonCurrencyUniqueSellersPrevious";
    NftCollectionRankingAttribute["Stats1hUsdAverageChange"] = "stats1hUsdAverageChange";
    NftCollectionRankingAttribute["Stats1hUsdAverageCurrent"] = "stats1hUsdAverageCurrent";
    NftCollectionRankingAttribute["Stats1hUsdAveragePrevious"] = "stats1hUsdAveragePrevious";
    NftCollectionRankingAttribute["Stats1hUsdCloseChange"] = "stats1hUsdCloseChange";
    NftCollectionRankingAttribute["Stats1hUsdCloseCurrent"] = "stats1hUsdCloseCurrent";
    NftCollectionRankingAttribute["Stats1hUsdClosePrevious"] = "stats1hUsdClosePrevious";
    NftCollectionRankingAttribute["Stats1hUsdHighestSaleChange"] = "stats1hUsdHighestSaleChange";
    NftCollectionRankingAttribute["Stats1hUsdHighestSaleCurrent"] = "stats1hUsdHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats1hUsdHighestSalePrevious"] = "stats1hUsdHighestSalePrevious";
    NftCollectionRankingAttribute["Stats1hUsdLowestSaleChange"] = "stats1hUsdLowestSaleChange";
    NftCollectionRankingAttribute["Stats1hUsdLowestSaleCurrent"] = "stats1hUsdLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats1hUsdLowestSalePrevious"] = "stats1hUsdLowestSalePrevious";
    NftCollectionRankingAttribute["Stats1hUsdOpenChange"] = "stats1hUsdOpenChange";
    NftCollectionRankingAttribute["Stats1hUsdOpenCurrent"] = "stats1hUsdOpenCurrent";
    NftCollectionRankingAttribute["Stats1hUsdOpenPrevious"] = "stats1hUsdOpenPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlendChange"] = "stats1hUsdVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlendCurrent"] = "stats1hUsdVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlendPrevious"] = "stats1hUsdVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlurChange"] = "stats1hUsdVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlurCurrent"] = "stats1hUsdVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlurPrevious"] = "stats1hUsdVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlurV2Change"] = "stats1hUsdVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlurV2Current"] = "stats1hUsdVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceBlurV2Previous"] = "stats1hUsdVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceCryptopunksChange"] = "stats1hUsdVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceCryptopunksCurrent"] = "stats1hUsdVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceCryptopunksPrevious"] = "stats1hUsdVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceGemChange"] = "stats1hUsdVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceGemCurrent"] = "stats1hUsdVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceGemPrevious"] = "stats1hUsdVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceLooksrareChange"] = "stats1hUsdVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceLooksrareCurrent"] = "stats1hUsdVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceLooksrarePrevious"] = "stats1hUsdVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceLooksrareV2Change"] = "stats1hUsdVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceLooksrareV2Current"] = "stats1hUsdVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceLooksrareV2Previous"] = "stats1hUsdVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceOpenseaChange"] = "stats1hUsdVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceOpenseaCurrent"] = "stats1hUsdVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceOpenseaPrevious"] = "stats1hUsdVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceOpenseaProChange"] = "stats1hUsdVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceOpenseaProCurrent"] = "stats1hUsdVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceOpenseaProPrevious"] = "stats1hUsdVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSeaportChange"] = "stats1hUsdVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSeaportCurrent"] = "stats1hUsdVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSeaportPrevious"] = "stats1hUsdVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSudoswapChange"] = "stats1hUsdVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSudoswapCurrent"] = "stats1hUsdVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSudoswapPrevious"] = "stats1hUsdVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSudoswapV2Change"] = "stats1hUsdVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSudoswapV2Current"] = "stats1hUsdVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceSudoswapV2Previous"] = "stats1hUsdVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceX2Y2Change"] = "stats1hUsdVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceX2Y2Current"] = "stats1hUsdVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats1hUsdVolumeByFillsourceX2Y2Previous"] = "stats1hUsdVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats1hUsdVolumeChange"] = "stats1hUsdVolumeChange";
    NftCollectionRankingAttribute["Stats1hUsdVolumeCurrent"] = "stats1hUsdVolumeCurrent";
    NftCollectionRankingAttribute["Stats1hUsdVolumePrevious"] = "stats1hUsdVolumePrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenAverageChange"] = "stats4hNetworkBaseTokenAverageChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenAverageCurrent"] = "stats4hNetworkBaseTokenAverageCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenAveragePrevious"] = "stats4hNetworkBaseTokenAveragePrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenCloseChange"] = "stats4hNetworkBaseTokenCloseChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenCloseCurrent"] = "stats4hNetworkBaseTokenCloseCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenClosePrevious"] = "stats4hNetworkBaseTokenClosePrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenHighestSaleChange"] = "stats4hNetworkBaseTokenHighestSaleChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenHighestSaleCurrent"] = "stats4hNetworkBaseTokenHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenHighestSalePrevious"] = "stats4hNetworkBaseTokenHighestSalePrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenLowestSaleChange"] = "stats4hNetworkBaseTokenLowestSaleChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenLowestSaleCurrent"] = "stats4hNetworkBaseTokenLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenLowestSalePrevious"] = "stats4hNetworkBaseTokenLowestSalePrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenOpenChange"] = "stats4hNetworkBaseTokenOpenChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenOpenCurrent"] = "stats4hNetworkBaseTokenOpenCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenOpenPrevious"] = "stats4hNetworkBaseTokenOpenPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlendChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlendCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlendPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlurChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlurCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlurPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlurV2Change"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlurV2Current"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceBlurV2Previous"] = "stats4hNetworkBaseTokenVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceCryptopunksChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceGemChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceGemCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceGemPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceLooksrareChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change"] = "stats4hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current"] = "stats4hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous"] = "stats4hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceOpenseaChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceOpenseaProChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSeaportChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSeaportCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSeaportPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSudoswapChange"] = "stats4hNetworkBaseTokenVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent"] = "stats4hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious"] = "stats4hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change"] = "stats4hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current"] = "stats4hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous"] = "stats4hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceX2Y2Change"] = "stats4hNetworkBaseTokenVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceX2Y2Current"] = "stats4hNetworkBaseTokenVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeByFillsourceX2Y2Previous"] = "stats4hNetworkBaseTokenVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeChange"] = "stats4hNetworkBaseTokenVolumeChange";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumeCurrent"] = "stats4hNetworkBaseTokenVolumeCurrent";
    NftCollectionRankingAttribute["Stats4hNetworkBaseTokenVolumePrevious"] = "stats4hNetworkBaseTokenVolumePrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyMintsChange"] = "stats4hNonCurrencyMintsChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyMintsCurrent"] = "stats4hNonCurrencyMintsCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyMintsPrevious"] = "stats4hNonCurrencyMintsPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencySalesChange"] = "stats4hNonCurrencySalesChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencySalesCurrent"] = "stats4hNonCurrencySalesCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencySalesPrevious"] = "stats4hNonCurrencySalesPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyTokensSoldChange"] = "stats4hNonCurrencyTokensSoldChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyTokensSoldCurrent"] = "stats4hNonCurrencyTokensSoldCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyTokensSoldPrevious"] = "stats4hNonCurrencyTokensSoldPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyTransfersChange"] = "stats4hNonCurrencyTransfersChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyTransfersCurrent"] = "stats4hNonCurrencyTransfersCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyTransfersPrevious"] = "stats4hNonCurrencyTransfersPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueBuyersChange"] = "stats4hNonCurrencyUniqueBuyersChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueBuyersCurrent"] = "stats4hNonCurrencyUniqueBuyersCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueBuyersPrevious"] = "stats4hNonCurrencyUniqueBuyersPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueMintersChange"] = "stats4hNonCurrencyUniqueMintersChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueMintersCurrent"] = "stats4hNonCurrencyUniqueMintersCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueMintersPrevious"] = "stats4hNonCurrencyUniqueMintersPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueSalesWalletsChange"] = "stats4hNonCurrencyUniqueSalesWalletsChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueSalesWalletsCurrent"] = "stats4hNonCurrencyUniqueSalesWalletsCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueSalesWalletsPrevious"] = "stats4hNonCurrencyUniqueSalesWalletsPrevious";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueSellersChange"] = "stats4hNonCurrencyUniqueSellersChange";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueSellersCurrent"] = "stats4hNonCurrencyUniqueSellersCurrent";
    NftCollectionRankingAttribute["Stats4hNonCurrencyUniqueSellersPrevious"] = "stats4hNonCurrencyUniqueSellersPrevious";
    NftCollectionRankingAttribute["Stats4hUsdAverageChange"] = "stats4hUsdAverageChange";
    NftCollectionRankingAttribute["Stats4hUsdAverageCurrent"] = "stats4hUsdAverageCurrent";
    NftCollectionRankingAttribute["Stats4hUsdAveragePrevious"] = "stats4hUsdAveragePrevious";
    NftCollectionRankingAttribute["Stats4hUsdCloseChange"] = "stats4hUsdCloseChange";
    NftCollectionRankingAttribute["Stats4hUsdCloseCurrent"] = "stats4hUsdCloseCurrent";
    NftCollectionRankingAttribute["Stats4hUsdClosePrevious"] = "stats4hUsdClosePrevious";
    NftCollectionRankingAttribute["Stats4hUsdHighestSaleChange"] = "stats4hUsdHighestSaleChange";
    NftCollectionRankingAttribute["Stats4hUsdHighestSaleCurrent"] = "stats4hUsdHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats4hUsdHighestSalePrevious"] = "stats4hUsdHighestSalePrevious";
    NftCollectionRankingAttribute["Stats4hUsdLowestSaleChange"] = "stats4hUsdLowestSaleChange";
    NftCollectionRankingAttribute["Stats4hUsdLowestSaleCurrent"] = "stats4hUsdLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats4hUsdLowestSalePrevious"] = "stats4hUsdLowestSalePrevious";
    NftCollectionRankingAttribute["Stats4hUsdOpenChange"] = "stats4hUsdOpenChange";
    NftCollectionRankingAttribute["Stats4hUsdOpenCurrent"] = "stats4hUsdOpenCurrent";
    NftCollectionRankingAttribute["Stats4hUsdOpenPrevious"] = "stats4hUsdOpenPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlendChange"] = "stats4hUsdVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlendCurrent"] = "stats4hUsdVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlendPrevious"] = "stats4hUsdVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlurChange"] = "stats4hUsdVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlurCurrent"] = "stats4hUsdVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlurPrevious"] = "stats4hUsdVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlurV2Change"] = "stats4hUsdVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlurV2Current"] = "stats4hUsdVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceBlurV2Previous"] = "stats4hUsdVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceCryptopunksChange"] = "stats4hUsdVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceCryptopunksCurrent"] = "stats4hUsdVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceCryptopunksPrevious"] = "stats4hUsdVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceGemChange"] = "stats4hUsdVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceGemCurrent"] = "stats4hUsdVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceGemPrevious"] = "stats4hUsdVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceLooksrareChange"] = "stats4hUsdVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceLooksrareCurrent"] = "stats4hUsdVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceLooksrarePrevious"] = "stats4hUsdVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceLooksrareV2Change"] = "stats4hUsdVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceLooksrareV2Current"] = "stats4hUsdVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceLooksrareV2Previous"] = "stats4hUsdVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceOpenseaChange"] = "stats4hUsdVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceOpenseaCurrent"] = "stats4hUsdVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceOpenseaPrevious"] = "stats4hUsdVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceOpenseaProChange"] = "stats4hUsdVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceOpenseaProCurrent"] = "stats4hUsdVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceOpenseaProPrevious"] = "stats4hUsdVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSeaportChange"] = "stats4hUsdVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSeaportCurrent"] = "stats4hUsdVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSeaportPrevious"] = "stats4hUsdVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSudoswapChange"] = "stats4hUsdVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSudoswapCurrent"] = "stats4hUsdVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSudoswapPrevious"] = "stats4hUsdVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSudoswapV2Change"] = "stats4hUsdVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSudoswapV2Current"] = "stats4hUsdVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceSudoswapV2Previous"] = "stats4hUsdVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceX2Y2Change"] = "stats4hUsdVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceX2Y2Current"] = "stats4hUsdVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats4hUsdVolumeByFillsourceX2Y2Previous"] = "stats4hUsdVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats4hUsdVolumeChange"] = "stats4hUsdVolumeChange";
    NftCollectionRankingAttribute["Stats4hUsdVolumeCurrent"] = "stats4hUsdVolumeCurrent";
    NftCollectionRankingAttribute["Stats4hUsdVolumePrevious"] = "stats4hUsdVolumePrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenAverageChange"] = "stats12hNetworkBaseTokenAverageChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenAverageCurrent"] = "stats12hNetworkBaseTokenAverageCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenAveragePrevious"] = "stats12hNetworkBaseTokenAveragePrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenCloseChange"] = "stats12hNetworkBaseTokenCloseChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenCloseCurrent"] = "stats12hNetworkBaseTokenCloseCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenClosePrevious"] = "stats12hNetworkBaseTokenClosePrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenHighestSaleChange"] = "stats12hNetworkBaseTokenHighestSaleChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenHighestSaleCurrent"] = "stats12hNetworkBaseTokenHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenHighestSalePrevious"] = "stats12hNetworkBaseTokenHighestSalePrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenLowestSaleChange"] = "stats12hNetworkBaseTokenLowestSaleChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenLowestSaleCurrent"] = "stats12hNetworkBaseTokenLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenLowestSalePrevious"] = "stats12hNetworkBaseTokenLowestSalePrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenOpenChange"] = "stats12hNetworkBaseTokenOpenChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenOpenCurrent"] = "stats12hNetworkBaseTokenOpenCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenOpenPrevious"] = "stats12hNetworkBaseTokenOpenPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlendChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlendCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlendPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlurChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlurCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlurPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlurV2Change"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlurV2Current"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceBlurV2Previous"] = "stats12hNetworkBaseTokenVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceCryptopunksChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceGemChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceGemCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceGemPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceLooksrareChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change"] = "stats12hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current"] = "stats12hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous"] = "stats12hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceOpenseaChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceOpenseaProChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSeaportChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSeaportCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSeaportPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSudoswapChange"] = "stats12hNetworkBaseTokenVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent"] = "stats12hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious"] = "stats12hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change"] = "stats12hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current"] = "stats12hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous"] = "stats12hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceX2Y2Change"] = "stats12hNetworkBaseTokenVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceX2Y2Current"] = "stats12hNetworkBaseTokenVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeByFillsourceX2Y2Previous"] = "stats12hNetworkBaseTokenVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeChange"] = "stats12hNetworkBaseTokenVolumeChange";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumeCurrent"] = "stats12hNetworkBaseTokenVolumeCurrent";
    NftCollectionRankingAttribute["Stats12hNetworkBaseTokenVolumePrevious"] = "stats12hNetworkBaseTokenVolumePrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyMintsChange"] = "stats12hNonCurrencyMintsChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyMintsCurrent"] = "stats12hNonCurrencyMintsCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyMintsPrevious"] = "stats12hNonCurrencyMintsPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencySalesChange"] = "stats12hNonCurrencySalesChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencySalesCurrent"] = "stats12hNonCurrencySalesCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencySalesPrevious"] = "stats12hNonCurrencySalesPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyTokensSoldChange"] = "stats12hNonCurrencyTokensSoldChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyTokensSoldCurrent"] = "stats12hNonCurrencyTokensSoldCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyTokensSoldPrevious"] = "stats12hNonCurrencyTokensSoldPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyTransfersChange"] = "stats12hNonCurrencyTransfersChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyTransfersCurrent"] = "stats12hNonCurrencyTransfersCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyTransfersPrevious"] = "stats12hNonCurrencyTransfersPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueBuyersChange"] = "stats12hNonCurrencyUniqueBuyersChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueBuyersCurrent"] = "stats12hNonCurrencyUniqueBuyersCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueBuyersPrevious"] = "stats12hNonCurrencyUniqueBuyersPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueMintersChange"] = "stats12hNonCurrencyUniqueMintersChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueMintersCurrent"] = "stats12hNonCurrencyUniqueMintersCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueMintersPrevious"] = "stats12hNonCurrencyUniqueMintersPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueSalesWalletsChange"] = "stats12hNonCurrencyUniqueSalesWalletsChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueSalesWalletsCurrent"] = "stats12hNonCurrencyUniqueSalesWalletsCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueSalesWalletsPrevious"] = "stats12hNonCurrencyUniqueSalesWalletsPrevious";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueSellersChange"] = "stats12hNonCurrencyUniqueSellersChange";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueSellersCurrent"] = "stats12hNonCurrencyUniqueSellersCurrent";
    NftCollectionRankingAttribute["Stats12hNonCurrencyUniqueSellersPrevious"] = "stats12hNonCurrencyUniqueSellersPrevious";
    NftCollectionRankingAttribute["Stats12hUsdAverageChange"] = "stats12hUsdAverageChange";
    NftCollectionRankingAttribute["Stats12hUsdAverageCurrent"] = "stats12hUsdAverageCurrent";
    NftCollectionRankingAttribute["Stats12hUsdAveragePrevious"] = "stats12hUsdAveragePrevious";
    NftCollectionRankingAttribute["Stats12hUsdCloseChange"] = "stats12hUsdCloseChange";
    NftCollectionRankingAttribute["Stats12hUsdCloseCurrent"] = "stats12hUsdCloseCurrent";
    NftCollectionRankingAttribute["Stats12hUsdClosePrevious"] = "stats12hUsdClosePrevious";
    NftCollectionRankingAttribute["Stats12hUsdHighestSaleChange"] = "stats12hUsdHighestSaleChange";
    NftCollectionRankingAttribute["Stats12hUsdHighestSaleCurrent"] = "stats12hUsdHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats12hUsdHighestSalePrevious"] = "stats12hUsdHighestSalePrevious";
    NftCollectionRankingAttribute["Stats12hUsdLowestSaleChange"] = "stats12hUsdLowestSaleChange";
    NftCollectionRankingAttribute["Stats12hUsdLowestSaleCurrent"] = "stats12hUsdLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats12hUsdLowestSalePrevious"] = "stats12hUsdLowestSalePrevious";
    NftCollectionRankingAttribute["Stats12hUsdOpenChange"] = "stats12hUsdOpenChange";
    NftCollectionRankingAttribute["Stats12hUsdOpenCurrent"] = "stats12hUsdOpenCurrent";
    NftCollectionRankingAttribute["Stats12hUsdOpenPrevious"] = "stats12hUsdOpenPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlendChange"] = "stats12hUsdVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlendCurrent"] = "stats12hUsdVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlendPrevious"] = "stats12hUsdVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlurChange"] = "stats12hUsdVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlurCurrent"] = "stats12hUsdVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlurPrevious"] = "stats12hUsdVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlurV2Change"] = "stats12hUsdVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlurV2Current"] = "stats12hUsdVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceBlurV2Previous"] = "stats12hUsdVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceCryptopunksChange"] = "stats12hUsdVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceCryptopunksCurrent"] = "stats12hUsdVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceCryptopunksPrevious"] = "stats12hUsdVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceGemChange"] = "stats12hUsdVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceGemCurrent"] = "stats12hUsdVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceGemPrevious"] = "stats12hUsdVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceLooksrareChange"] = "stats12hUsdVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceLooksrareCurrent"] = "stats12hUsdVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceLooksrarePrevious"] = "stats12hUsdVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceLooksrareV2Change"] = "stats12hUsdVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceLooksrareV2Current"] = "stats12hUsdVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceLooksrareV2Previous"] = "stats12hUsdVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceOpenseaChange"] = "stats12hUsdVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceOpenseaCurrent"] = "stats12hUsdVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceOpenseaPrevious"] = "stats12hUsdVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceOpenseaProChange"] = "stats12hUsdVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceOpenseaProCurrent"] = "stats12hUsdVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceOpenseaProPrevious"] = "stats12hUsdVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSeaportChange"] = "stats12hUsdVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSeaportCurrent"] = "stats12hUsdVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSeaportPrevious"] = "stats12hUsdVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSudoswapChange"] = "stats12hUsdVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSudoswapCurrent"] = "stats12hUsdVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSudoswapPrevious"] = "stats12hUsdVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSudoswapV2Change"] = "stats12hUsdVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSudoswapV2Current"] = "stats12hUsdVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceSudoswapV2Previous"] = "stats12hUsdVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceX2Y2Change"] = "stats12hUsdVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceX2Y2Current"] = "stats12hUsdVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats12hUsdVolumeByFillsourceX2Y2Previous"] = "stats12hUsdVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats12hUsdVolumeChange"] = "stats12hUsdVolumeChange";
    NftCollectionRankingAttribute["Stats12hUsdVolumeCurrent"] = "stats12hUsdVolumeCurrent";
    NftCollectionRankingAttribute["Stats12hUsdVolumePrevious"] = "stats12hUsdVolumePrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenAverageChange"] = "stats24hNetworkBaseTokenAverageChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenAverageCurrent"] = "stats24hNetworkBaseTokenAverageCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenAveragePrevious"] = "stats24hNetworkBaseTokenAveragePrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenCloseChange"] = "stats24hNetworkBaseTokenCloseChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenCloseCurrent"] = "stats24hNetworkBaseTokenCloseCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenClosePrevious"] = "stats24hNetworkBaseTokenClosePrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenHighestSaleChange"] = "stats24hNetworkBaseTokenHighestSaleChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenHighestSaleCurrent"] = "stats24hNetworkBaseTokenHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenHighestSalePrevious"] = "stats24hNetworkBaseTokenHighestSalePrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenLowestSaleChange"] = "stats24hNetworkBaseTokenLowestSaleChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenLowestSaleCurrent"] = "stats24hNetworkBaseTokenLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenLowestSalePrevious"] = "stats24hNetworkBaseTokenLowestSalePrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenOpenChange"] = "stats24hNetworkBaseTokenOpenChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenOpenCurrent"] = "stats24hNetworkBaseTokenOpenCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenOpenPrevious"] = "stats24hNetworkBaseTokenOpenPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlendChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlendCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlendPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlurChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlurCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlurPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlurV2Change"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlurV2Current"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceBlurV2Previous"] = "stats24hNetworkBaseTokenVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceCryptopunksChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceGemChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceGemCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceGemPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceLooksrareChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change"] = "stats24hNetworkBaseTokenVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current"] = "stats24hNetworkBaseTokenVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous"] = "stats24hNetworkBaseTokenVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceOpenseaChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceOpenseaProChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSeaportChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSeaportCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSeaportPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSudoswapChange"] = "stats24hNetworkBaseTokenVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent"] = "stats24hNetworkBaseTokenVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious"] = "stats24hNetworkBaseTokenVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change"] = "stats24hNetworkBaseTokenVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current"] = "stats24hNetworkBaseTokenVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous"] = "stats24hNetworkBaseTokenVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceX2Y2Change"] = "stats24hNetworkBaseTokenVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceX2Y2Current"] = "stats24hNetworkBaseTokenVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeByFillsourceX2Y2Previous"] = "stats24hNetworkBaseTokenVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeChange"] = "stats24hNetworkBaseTokenVolumeChange";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumeCurrent"] = "stats24hNetworkBaseTokenVolumeCurrent";
    NftCollectionRankingAttribute["Stats24hNetworkBaseTokenVolumePrevious"] = "stats24hNetworkBaseTokenVolumePrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyMintsChange"] = "stats24hNonCurrencyMintsChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyMintsCurrent"] = "stats24hNonCurrencyMintsCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyMintsPrevious"] = "stats24hNonCurrencyMintsPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencySalesChange"] = "stats24hNonCurrencySalesChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencySalesCurrent"] = "stats24hNonCurrencySalesCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencySalesPrevious"] = "stats24hNonCurrencySalesPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyTokensSoldChange"] = "stats24hNonCurrencyTokensSoldChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyTokensSoldCurrent"] = "stats24hNonCurrencyTokensSoldCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyTokensSoldPrevious"] = "stats24hNonCurrencyTokensSoldPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyTransfersChange"] = "stats24hNonCurrencyTransfersChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyTransfersCurrent"] = "stats24hNonCurrencyTransfersCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyTransfersPrevious"] = "stats24hNonCurrencyTransfersPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueBuyersChange"] = "stats24hNonCurrencyUniqueBuyersChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueBuyersCurrent"] = "stats24hNonCurrencyUniqueBuyersCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueBuyersPrevious"] = "stats24hNonCurrencyUniqueBuyersPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueMintersChange"] = "stats24hNonCurrencyUniqueMintersChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueMintersCurrent"] = "stats24hNonCurrencyUniqueMintersCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueMintersPrevious"] = "stats24hNonCurrencyUniqueMintersPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueSalesWalletsChange"] = "stats24hNonCurrencyUniqueSalesWalletsChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueSalesWalletsCurrent"] = "stats24hNonCurrencyUniqueSalesWalletsCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueSalesWalletsPrevious"] = "stats24hNonCurrencyUniqueSalesWalletsPrevious";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueSellersChange"] = "stats24hNonCurrencyUniqueSellersChange";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueSellersCurrent"] = "stats24hNonCurrencyUniqueSellersCurrent";
    NftCollectionRankingAttribute["Stats24hNonCurrencyUniqueSellersPrevious"] = "stats24hNonCurrencyUniqueSellersPrevious";
    NftCollectionRankingAttribute["Stats24hUsdAverageChange"] = "stats24hUsdAverageChange";
    NftCollectionRankingAttribute["Stats24hUsdAverageCurrent"] = "stats24hUsdAverageCurrent";
    NftCollectionRankingAttribute["Stats24hUsdAveragePrevious"] = "stats24hUsdAveragePrevious";
    NftCollectionRankingAttribute["Stats24hUsdCloseChange"] = "stats24hUsdCloseChange";
    NftCollectionRankingAttribute["Stats24hUsdCloseCurrent"] = "stats24hUsdCloseCurrent";
    NftCollectionRankingAttribute["Stats24hUsdClosePrevious"] = "stats24hUsdClosePrevious";
    NftCollectionRankingAttribute["Stats24hUsdHighestSaleChange"] = "stats24hUsdHighestSaleChange";
    NftCollectionRankingAttribute["Stats24hUsdHighestSaleCurrent"] = "stats24hUsdHighestSaleCurrent";
    NftCollectionRankingAttribute["Stats24hUsdHighestSalePrevious"] = "stats24hUsdHighestSalePrevious";
    NftCollectionRankingAttribute["Stats24hUsdLowestSaleChange"] = "stats24hUsdLowestSaleChange";
    NftCollectionRankingAttribute["Stats24hUsdLowestSaleCurrent"] = "stats24hUsdLowestSaleCurrent";
    NftCollectionRankingAttribute["Stats24hUsdLowestSalePrevious"] = "stats24hUsdLowestSalePrevious";
    NftCollectionRankingAttribute["Stats24hUsdOpenChange"] = "stats24hUsdOpenChange";
    NftCollectionRankingAttribute["Stats24hUsdOpenCurrent"] = "stats24hUsdOpenCurrent";
    NftCollectionRankingAttribute["Stats24hUsdOpenPrevious"] = "stats24hUsdOpenPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlendChange"] = "stats24hUsdVolumeByFillsourceBlendChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlendCurrent"] = "stats24hUsdVolumeByFillsourceBlendCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlendPrevious"] = "stats24hUsdVolumeByFillsourceBlendPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlurChange"] = "stats24hUsdVolumeByFillsourceBlurChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlurCurrent"] = "stats24hUsdVolumeByFillsourceBlurCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlurPrevious"] = "stats24hUsdVolumeByFillsourceBlurPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlurV2Change"] = "stats24hUsdVolumeByFillsourceBlurV2Change";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlurV2Current"] = "stats24hUsdVolumeByFillsourceBlurV2Current";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceBlurV2Previous"] = "stats24hUsdVolumeByFillsourceBlurV2Previous";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceCryptopunksChange"] = "stats24hUsdVolumeByFillsourceCryptopunksChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceCryptopunksCurrent"] = "stats24hUsdVolumeByFillsourceCryptopunksCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceCryptopunksPrevious"] = "stats24hUsdVolumeByFillsourceCryptopunksPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceGemChange"] = "stats24hUsdVolumeByFillsourceGemChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceGemCurrent"] = "stats24hUsdVolumeByFillsourceGemCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceGemPrevious"] = "stats24hUsdVolumeByFillsourceGemPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceLooksrareChange"] = "stats24hUsdVolumeByFillsourceLooksrareChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceLooksrareCurrent"] = "stats24hUsdVolumeByFillsourceLooksrareCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceLooksrarePrevious"] = "stats24hUsdVolumeByFillsourceLooksrarePrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceLooksrareV2Change"] = "stats24hUsdVolumeByFillsourceLooksrareV2Change";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceLooksrareV2Current"] = "stats24hUsdVolumeByFillsourceLooksrareV2Current";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceLooksrareV2Previous"] = "stats24hUsdVolumeByFillsourceLooksrareV2Previous";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceOpenseaChange"] = "stats24hUsdVolumeByFillsourceOpenseaChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceOpenseaCurrent"] = "stats24hUsdVolumeByFillsourceOpenseaCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceOpenseaPrevious"] = "stats24hUsdVolumeByFillsourceOpenseaPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceOpenseaProChange"] = "stats24hUsdVolumeByFillsourceOpenseaProChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceOpenseaProCurrent"] = "stats24hUsdVolumeByFillsourceOpenseaProCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceOpenseaProPrevious"] = "stats24hUsdVolumeByFillsourceOpenseaProPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSeaportChange"] = "stats24hUsdVolumeByFillsourceSeaportChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSeaportCurrent"] = "stats24hUsdVolumeByFillsourceSeaportCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSeaportPrevious"] = "stats24hUsdVolumeByFillsourceSeaportPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSudoswapChange"] = "stats24hUsdVolumeByFillsourceSudoswapChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSudoswapCurrent"] = "stats24hUsdVolumeByFillsourceSudoswapCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSudoswapPrevious"] = "stats24hUsdVolumeByFillsourceSudoswapPrevious";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSudoswapV2Change"] = "stats24hUsdVolumeByFillsourceSudoswapV2Change";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSudoswapV2Current"] = "stats24hUsdVolumeByFillsourceSudoswapV2Current";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceSudoswapV2Previous"] = "stats24hUsdVolumeByFillsourceSudoswapV2Previous";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceX2Y2Change"] = "stats24hUsdVolumeByFillsourceX2Y2Change";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceX2Y2Current"] = "stats24hUsdVolumeByFillsourceX2Y2Current";
    NftCollectionRankingAttribute["Stats24hUsdVolumeByFillsourceX2Y2Previous"] = "stats24hUsdVolumeByFillsourceX2Y2Previous";
    NftCollectionRankingAttribute["Stats24hUsdVolumeChange"] = "stats24hUsdVolumeChange";
    NftCollectionRankingAttribute["Stats24hUsdVolumeCurrent"] = "stats24hUsdVolumeCurrent";
    NftCollectionRankingAttribute["Stats24hUsdVolumePrevious"] = "stats24hUsdVolumePrevious";
    NftCollectionRankingAttribute["TotalSupply"] = "totalSupply";
})(NftCollectionRankingAttribute || (exports.NftCollectionRankingAttribute = NftCollectionRankingAttribute = {}));
/** The duration used to rank NFTs. */
var NftCollectionsLeaderboardDuration;
(function (NftCollectionsLeaderboardDuration) {
    NftCollectionsLeaderboardDuration["Day1"] = "day1";
    NftCollectionsLeaderboardDuration["Day30"] = "day30";
    NftCollectionsLeaderboardDuration["Hour1"] = "hour1";
    NftCollectionsLeaderboardDuration["Hour4"] = "hour4";
    NftCollectionsLeaderboardDuration["Hour12"] = "hour12";
    NftCollectionsLeaderboardDuration["Min15"] = "min15";
    NftCollectionsLeaderboardDuration["Week1"] = "week1";
})(NftCollectionsLeaderboardDuration || (exports.NftCollectionsLeaderboardDuration = NftCollectionsLeaderboardDuration = {}));
/** The attribute used to rank NFT collections. */
var NftCollectionsLeaderboardMetric;
(function (NftCollectionsLeaderboardMetric) {
    NftCollectionsLeaderboardMetric["Buyers"] = "buyers";
    NftCollectionsLeaderboardMetric["BuyersGain"] = "buyersGain";
    NftCollectionsLeaderboardMetric["Mints"] = "mints";
    NftCollectionsLeaderboardMetric["MintsGain"] = "mintsGain";
    NftCollectionsLeaderboardMetric["Sales"] = "sales";
    NftCollectionsLeaderboardMetric["SalesGain"] = "salesGain";
    NftCollectionsLeaderboardMetric["Sellers"] = "sellers";
    NftCollectionsLeaderboardMetric["SellersGain"] = "sellersGain";
    NftCollectionsLeaderboardMetric["TokensSold"] = "tokensSold";
    NftCollectionsLeaderboardMetric["TokensSoldGain"] = "tokensSoldGain";
    NftCollectionsLeaderboardMetric["VolumeBase"] = "volumeBase";
    NftCollectionsLeaderboardMetric["VolumeBaseGain"] = "volumeBaseGain";
    NftCollectionsLeaderboardMetric["VolumeUsd"] = "volumeUsd";
    NftCollectionsLeaderboardMetric["VolumeUsdGain"] = "volumeUsdGain";
})(NftCollectionsLeaderboardMetric || (exports.NftCollectionsLeaderboardMetric = NftCollectionsLeaderboardMetric = {}));
/** Token standards. */
var NftContractErcType;
(function (NftContractErcType) {
    NftContractErcType["Erc721"] = "ERC721";
    NftContractErcType["Erc721Enumerable"] = "ERC721Enumerable";
    NftContractErcType["Erc721Metadata"] = "ERC721Metadata";
    NftContractErcType["Erc1155"] = "ERC1155";
    NftContractErcType["Erc1155Metadata"] = "ERC1155Metadata";
    NftContractErcType["Unsupported"] = "Unsupported";
})(NftContractErcType || (exports.NftContractErcType = NftContractErcType = {}));
/** The direction of the nft sale event. */
var NftEventOrderDirection;
(function (NftEventOrderDirection) {
    NftEventOrderDirection["Buy"] = "BUY";
    NftEventOrderDirection["OfferAccepted"] = "OFFER_ACCEPTED";
    NftEventOrderDirection["Sell"] = "SELL";
})(NftEventOrderDirection || (exports.NftEventOrderDirection = NftEventOrderDirection = {}));
/** The type of item involved in the trade. */
var NftEventTradeItemType;
(function (NftEventTradeItemType) {
    NftEventTradeItemType["Nft"] = "NFT";
    NftEventTradeItemType["Token"] = "TOKEN";
})(NftEventTradeItemType || (exports.NftEventTradeItemType = NftEventTradeItemType = {}));
/** NFT marketplaces. */
var NftExchange;
(function (NftExchange) {
    NftExchange["Alto"] = "ALTO";
    NftExchange["Bitkeep"] = "BITKEEP";
    NftExchange["Blend"] = "BLEND";
    NftExchange["Blur"] = "BLUR";
    NftExchange["Blurv2"] = "BLURV2";
    NftExchange["Cryptopunks"] = "CRYPTOPUNKS";
    NftExchange["Joepegs"] = "JOEPEGS";
    NftExchange["Looksrare"] = "LOOKSRARE";
    NftExchange["Looksrarev2"] = "LOOKSRAREV2";
    NftExchange["Minted"] = "MINTED";
    NftExchange["Opensea"] = "OPENSEA";
    NftExchange["Playdapp"] = "PLAYDAPP";
    NftExchange["Provenant"] = "PROVENANT";
    NftExchange["Quixotic"] = "QUIXOTIC";
    NftExchange["Seaport"] = "SEAPORT";
    NftExchange["Stratos"] = "STRATOS";
    NftExchange["Sudoswapammv2"] = "SUDOSWAPAMMV2";
    NftExchange["Sudoswapv2"] = "SUDOSWAPV2";
    NftExchange["Tofunft"] = "TOFUNFT";
    NftExchange["Treasure"] = "TREASURE";
    NftExchange["X2Y2"] = "X2Y2";
    NftExchange["Zeroexv3"] = "ZEROEXV3";
    NftExchange["Zeroexv4"] = "ZEROEXV4";
})(NftExchange || (exports.NftExchange = NftExchange = {}));
/** The attribute used to rank NFT collections. */
var NftPoolCollectionRankingAttribute;
(function (NftPoolCollectionRankingAttribute) {
    NftPoolCollectionRankingAttribute["BalanceNbt"] = "balanceNBT";
    NftPoolCollectionRankingAttribute["BalanceUsd"] = "balanceUSD";
    NftPoolCollectionRankingAttribute["ExpenseNbt24"] = "expenseNBT24";
    NftPoolCollectionRankingAttribute["ExpenseNbtAll"] = "expenseNBTAll";
    NftPoolCollectionRankingAttribute["ExpenseUsd24"] = "expenseUSD24";
    NftPoolCollectionRankingAttribute["ExpenseUsdAll"] = "expenseUSDAll";
    NftPoolCollectionRankingAttribute["FloorNbt"] = "floorNBT";
    NftPoolCollectionRankingAttribute["FloorUsd"] = "floorUSD";
    NftPoolCollectionRankingAttribute["HighPriceNbt24"] = "highPriceNBT24";
    NftPoolCollectionRankingAttribute["HighPriceNbtAll"] = "highPriceNBTAll";
    NftPoolCollectionRankingAttribute["HighPriceUsd24"] = "highPriceUSD24";
    NftPoolCollectionRankingAttribute["HighPriceUsdAll"] = "highPriceUSDAll";
    NftPoolCollectionRankingAttribute["LowPriceNbt24"] = "lowPriceNBT24";
    NftPoolCollectionRankingAttribute["LowPriceNbtAll"] = "lowPriceNBTAll";
    NftPoolCollectionRankingAttribute["LowPriceUsd24"] = "lowPriceUSD24";
    NftPoolCollectionRankingAttribute["LowPriceUsdAll"] = "lowPriceUSDAll";
    NftPoolCollectionRankingAttribute["NftBalance"] = "nftBalance";
    NftPoolCollectionRankingAttribute["NftVolume24"] = "nftVolume24";
    NftPoolCollectionRankingAttribute["NftVolumeAll"] = "nftVolumeAll";
    NftPoolCollectionRankingAttribute["NftsBought24"] = "nftsBought24";
    NftPoolCollectionRankingAttribute["NftsBoughtAll"] = "nftsBoughtAll";
    NftPoolCollectionRankingAttribute["NftsSold24"] = "nftsSold24";
    NftPoolCollectionRankingAttribute["NftsSoldAll"] = "nftsSoldAll";
    NftPoolCollectionRankingAttribute["OfferNbt"] = "offerNBT";
    NftPoolCollectionRankingAttribute["OfferUsd"] = "offerUSD";
    NftPoolCollectionRankingAttribute["PoolFeesNbt24"] = "poolFeesNBT24";
    NftPoolCollectionRankingAttribute["PoolFeesNbtAll"] = "poolFeesNBTAll";
    NftPoolCollectionRankingAttribute["PoolFeesUsd24"] = "poolFeesUSD24";
    NftPoolCollectionRankingAttribute["PoolFeesUsdAll"] = "poolFeesUSDAll";
    NftPoolCollectionRankingAttribute["ProtocolFeesNbt24"] = "protocolFeesNBT24";
    NftPoolCollectionRankingAttribute["ProtocolFeesNbtAll"] = "protocolFeesNBTAll";
    NftPoolCollectionRankingAttribute["ProtocolFeesUsd24"] = "protocolFeesUSD24";
    NftPoolCollectionRankingAttribute["ProtocolFeesUsdAll"] = "protocolFeesUSDAll";
    NftPoolCollectionRankingAttribute["RevenueNbt24"] = "revenueNBT24";
    NftPoolCollectionRankingAttribute["RevenueNbtAll"] = "revenueNBTAll";
    NftPoolCollectionRankingAttribute["RevenueUsd24"] = "revenueUSD24";
    NftPoolCollectionRankingAttribute["RevenueUsdAll"] = "revenueUSDAll";
    NftPoolCollectionRankingAttribute["TotalSupply"] = "totalSupply";
    NftPoolCollectionRankingAttribute["VolumeNbt24"] = "volumeNBT24";
    NftPoolCollectionRankingAttribute["VolumeNbtAll"] = "volumeNBTAll";
    NftPoolCollectionRankingAttribute["VolumeUsd24"] = "volumeUSD24";
    NftPoolCollectionRankingAttribute["VolumeUsdAll"] = "volumeUSDAll";
})(NftPoolCollectionRankingAttribute || (exports.NftPoolCollectionRankingAttribute = NftPoolCollectionRankingAttribute = {}));
/** The NFT pool contract version. */
var NftPoolContractVersion;
(function (NftPoolContractVersion) {
    NftPoolContractVersion["SudoswapV1"] = "SUDOSWAP_V1";
    NftPoolContractVersion["SudoswapV2"] = "SUDOSWAP_V2";
})(NftPoolContractVersion || (exports.NftPoolContractVersion = NftPoolContractVersion = {}));
/** The type of an NFT pool event. */
var NftPoolEventType;
(function (NftPoolEventType) {
    NftPoolEventType["AssetRecipientChange"] = "ASSET_RECIPIENT_CHANGE";
    NftPoolEventType["DeltaUpdate"] = "DELTA_UPDATE";
    NftPoolEventType["FeeUpdate"] = "FEE_UPDATE";
    NftPoolEventType["NewPool"] = "NEW_POOL";
    NftPoolEventType["NewPoolV2"] = "NEW_POOL_V2";
    NftPoolEventType["NftDeposit"] = "NFT_DEPOSIT";
    NftPoolEventType["NftDepositV2"] = "NFT_DEPOSIT_V2";
    NftPoolEventType["NftWithdrawal"] = "NFT_WITHDRAWAL";
    NftPoolEventType["NftWithdrawalV2"] = "NFT_WITHDRAWAL_V2";
    NftPoolEventType["OwnershipTransferred"] = "OWNERSHIP_TRANSFERRED";
    NftPoolEventType["SpotPriceUpdate"] = "SPOT_PRICE_UPDATE";
    NftPoolEventType["SpotPriceUpdateV2"] = "SPOT_PRICE_UPDATE_V2";
    NftPoolEventType["SwapNftInPool"] = "SWAP_NFT_IN_POOL";
    NftPoolEventType["SwapNftInPoolV2"] = "SWAP_NFT_IN_POOL_V2";
    NftPoolEventType["SwapNftOutPool"] = "SWAP_NFT_OUT_POOL";
    NftPoolEventType["SwapNftOutPoolV2"] = "SWAP_NFT_OUT_POOL_V2";
    NftPoolEventType["TokenDeposit"] = "TOKEN_DEPOSIT";
    NftPoolEventType["TokenDepositV2"] = "TOKEN_DEPOSIT_V2";
    NftPoolEventType["TokenWithdrawal"] = "TOKEN_WITHDRAWAL";
    NftPoolEventType["TokenWithdrawalV2"] = "TOKEN_WITHDRAWAL_V2";
})(NftPoolEventType || (exports.NftPoolEventType = NftPoolEventType = {}));
/** The attribute used to rank NFT pools. */
var NftPoolRankingAttribute;
(function (NftPoolRankingAttribute) {
    NftPoolRankingAttribute["BalanceNbt"] = "balanceNBT";
    NftPoolRankingAttribute["BalanceUsd"] = "balanceUSD";
    NftPoolRankingAttribute["ExpenseNbt24"] = "expenseNBT24";
    NftPoolRankingAttribute["ExpenseNbtAll"] = "expenseNBTAll";
    NftPoolRankingAttribute["ExpenseUsd24"] = "expenseUSD24";
    NftPoolRankingAttribute["ExpenseUsdAll"] = "expenseUSDAll";
    NftPoolRankingAttribute["NftBalance"] = "nftBalance";
    NftPoolRankingAttribute["NftVolume24"] = "nftVolume24";
    NftPoolRankingAttribute["NftVolumeAll"] = "nftVolumeAll";
    NftPoolRankingAttribute["NftsBought24"] = "nftsBought24";
    NftPoolRankingAttribute["NftsBoughtAll"] = "nftsBoughtAll";
    NftPoolRankingAttribute["NftsSold24"] = "nftsSold24";
    NftPoolRankingAttribute["NftsSoldAll"] = "nftsSoldAll";
    NftPoolRankingAttribute["OfferNbt"] = "offerNBT";
    NftPoolRankingAttribute["OfferUsd"] = "offerUSD";
    NftPoolRankingAttribute["PoolFeesNbt24"] = "poolFeesNBT24";
    NftPoolRankingAttribute["PoolFeesNbtAll"] = "poolFeesNBTAll";
    NftPoolRankingAttribute["PoolFeesUsd24"] = "poolFeesUSD24";
    NftPoolRankingAttribute["PoolFeesUsdAll"] = "poolFeesUSDAll";
    NftPoolRankingAttribute["ProtocolFeesNbt24"] = "protocolFeesNBT24";
    NftPoolRankingAttribute["ProtocolFeesNbtAll"] = "protocolFeesNBTAll";
    NftPoolRankingAttribute["ProtocolFeesUsd24"] = "protocolFeesUSD24";
    NftPoolRankingAttribute["ProtocolFeesUsdAll"] = "protocolFeesUSDAll";
    NftPoolRankingAttribute["RevenueNbt24"] = "revenueNBT24";
    NftPoolRankingAttribute["RevenueNbtAll"] = "revenueNBTAll";
    NftPoolRankingAttribute["RevenueUsd24"] = "revenueUSD24";
    NftPoolRankingAttribute["RevenueUsdAll"] = "revenueUSDAll";
    NftPoolRankingAttribute["SellNbt"] = "sellNBT";
    NftPoolRankingAttribute["SellUsd"] = "sellUSD";
    NftPoolRankingAttribute["VolumeNbt24"] = "volumeNBT24";
    NftPoolRankingAttribute["VolumeNbtAll"] = "volumeNBTAll";
    NftPoolRankingAttribute["VolumeUsd24"] = "volumeUSD24";
    NftPoolRankingAttribute["VolumeUsdAll"] = "volumeUSDAll";
})(NftPoolRankingAttribute || (exports.NftPoolRankingAttribute = NftPoolRankingAttribute = {}));
/** The pool type selected by the pool creator. */
var NftPoolType;
(function (NftPoolType) {
    NftPoolType["Buy"] = "BUY";
    NftPoolType["BuyAndSell"] = "BUY_AND_SELL";
    NftPoolType["Sell"] = "SELL";
})(NftPoolType || (exports.NftPoolType = NftPoolType = {}));
/** The level of NFTs to search. */
var NftSearchable;
(function (NftSearchable) {
    NftSearchable["Asset"] = "Asset";
    NftSearchable["Collection"] = "Collection";
})(NftSearchable || (exports.NftSearchable = NftSearchable = {}));
var OrderState;
(function (OrderState) {
    OrderState["Completed"] = "COMPLETED";
    OrderState["Created"] = "CREATED";
    OrderState["Expired"] = "EXPIRED";
    OrderState["Failed"] = "FAILED";
    OrderState["Pending"] = "PENDING";
})(OrderState || (exports.OrderState = OrderState = {}));
var OrderType;
(function (OrderType) {
    OrderType["Limit"] = "LIMIT";
    OrderType["Manual"] = "MANUAL";
})(OrderType || (exports.OrderType = OrderType = {}));
/** The attribute used to rank tokens. */
var PairRankingAttribute;
(function (PairRankingAttribute) {
    PairRankingAttribute["BuyCount1"] = "buyCount1";
    PairRankingAttribute["BuyCount4"] = "buyCount4";
    PairRankingAttribute["BuyCount12"] = "buyCount12";
    PairRankingAttribute["BuyCount24"] = "buyCount24";
    PairRankingAttribute["CreatedAt"] = "createdAt";
    PairRankingAttribute["HighPrice1"] = "highPrice1";
    PairRankingAttribute["HighPrice4"] = "highPrice4";
    PairRankingAttribute["HighPrice12"] = "highPrice12";
    PairRankingAttribute["HighPrice24"] = "highPrice24";
    PairRankingAttribute["LastTransaction"] = "lastTransaction";
    PairRankingAttribute["Liquidity"] = "liquidity";
    PairRankingAttribute["LockedLiquidityPercentage"] = "lockedLiquidityPercentage";
    PairRankingAttribute["LowPrice1"] = "lowPrice1";
    PairRankingAttribute["LowPrice4"] = "lowPrice4";
    PairRankingAttribute["LowPrice12"] = "lowPrice12";
    PairRankingAttribute["LowPrice24"] = "lowPrice24";
    PairRankingAttribute["MarketCap"] = "marketCap";
    PairRankingAttribute["Price"] = "price";
    PairRankingAttribute["PriceChange1"] = "priceChange1";
    PairRankingAttribute["PriceChange4"] = "priceChange4";
    PairRankingAttribute["PriceChange12"] = "priceChange12";
    PairRankingAttribute["PriceChange24"] = "priceChange24";
    PairRankingAttribute["SellCount1"] = "sellCount1";
    PairRankingAttribute["SellCount4"] = "sellCount4";
    PairRankingAttribute["SellCount12"] = "sellCount12";
    PairRankingAttribute["SellCount24"] = "sellCount24";
    PairRankingAttribute["TrendingScore"] = "trendingScore";
    PairRankingAttribute["TrendingScore1"] = "trendingScore1";
    PairRankingAttribute["TrendingScore4"] = "trendingScore4";
    PairRankingAttribute["TrendingScore12"] = "trendingScore12";
    PairRankingAttribute["TrendingScore24"] = "trendingScore24";
    PairRankingAttribute["TxnCount1"] = "txnCount1";
    PairRankingAttribute["TxnCount4"] = "txnCount4";
    PairRankingAttribute["TxnCount12"] = "txnCount12";
    PairRankingAttribute["TxnCount24"] = "txnCount24";
    PairRankingAttribute["UniqueBuys1"] = "uniqueBuys1";
    PairRankingAttribute["UniqueBuys4"] = "uniqueBuys4";
    PairRankingAttribute["UniqueBuys12"] = "uniqueBuys12";
    PairRankingAttribute["UniqueBuys24"] = "uniqueBuys24";
    PairRankingAttribute["UniqueSells1"] = "uniqueSells1";
    PairRankingAttribute["UniqueSells4"] = "uniqueSells4";
    PairRankingAttribute["UniqueSells12"] = "uniqueSells12";
    PairRankingAttribute["UniqueSells24"] = "uniqueSells24";
    PairRankingAttribute["UniqueTransactions1"] = "uniqueTransactions1";
    PairRankingAttribute["UniqueTransactions4"] = "uniqueTransactions4";
    PairRankingAttribute["UniqueTransactions12"] = "uniqueTransactions12";
    PairRankingAttribute["UniqueTransactions24"] = "uniqueTransactions24";
    PairRankingAttribute["VolumeChange1"] = "volumeChange1";
    PairRankingAttribute["VolumeChange4"] = "volumeChange4";
    PairRankingAttribute["VolumeChange12"] = "volumeChange12";
    PairRankingAttribute["VolumeChange24"] = "volumeChange24";
    PairRankingAttribute["VolumeUsd1"] = "volumeUSD1";
    PairRankingAttribute["VolumeUsd4"] = "volumeUSD4";
    PairRankingAttribute["VolumeUsd12"] = "volumeUSD12";
    PairRankingAttribute["VolumeUsd24"] = "volumeUSD24";
})(PairRankingAttribute || (exports.PairRankingAttribute = PairRankingAttribute = {}));
/** The Parallel asset card type. */
var ParallelAssetMatcherCardType;
(function (ParallelAssetMatcherCardType) {
    ParallelAssetMatcherCardType["Effect"] = "Effect";
    ParallelAssetMatcherCardType["Paragon"] = "Paragon";
    ParallelAssetMatcherCardType["Relic"] = "Relic";
    ParallelAssetMatcherCardType["Unit"] = "Unit";
    ParallelAssetMatcherCardType["Upgrade"] = "Upgrade";
})(ParallelAssetMatcherCardType || (exports.ParallelAssetMatcherCardType = ParallelAssetMatcherCardType = {}));
/** The Parallel asset class. */
var ParallelAssetMatcherClass;
(function (ParallelAssetMatcherClass) {
    ParallelAssetMatcherClass["ArtCard"] = "ArtCard";
    ParallelAssetMatcherClass["Asset"] = "Asset";
    ParallelAssetMatcherClass["CardBack"] = "CardBack";
    ParallelAssetMatcherClass["Fe"] = "FE";
    ParallelAssetMatcherClass["Masterpiece"] = "Masterpiece";
    ParallelAssetMatcherClass["Pl"] = "PL";
    ParallelAssetMatcherClass["Se"] = "SE";
})(ParallelAssetMatcherClass || (exports.ParallelAssetMatcherClass = ParallelAssetMatcherClass = {}));
/** The Parallel stream of evolution. */
var ParallelAssetMatcherParallel;
(function (ParallelAssetMatcherParallel) {
    ParallelAssetMatcherParallel["Augencore"] = "Augencore";
    ParallelAssetMatcherParallel["Earthen"] = "Earthen";
    ParallelAssetMatcherParallel["Kathari"] = "Kathari";
    ParallelAssetMatcherParallel["Marcolian"] = "Marcolian";
    ParallelAssetMatcherParallel["Shroud"] = "Shroud";
    ParallelAssetMatcherParallel["Universal"] = "Universal";
    ParallelAssetMatcherParallel["UnknownOrigins"] = "UnknownOrigins";
})(ParallelAssetMatcherParallel || (exports.ParallelAssetMatcherParallel = ParallelAssetMatcherParallel = {}));
/** The Parallel asset rarity. */
var ParallelAssetMatcherRarity;
(function (ParallelAssetMatcherRarity) {
    ParallelAssetMatcherRarity["Common"] = "Common";
    ParallelAssetMatcherRarity["Legendary"] = "Legendary";
    ParallelAssetMatcherRarity["Prime"] = "Prime";
    ParallelAssetMatcherRarity["Rare"] = "Rare";
    ParallelAssetMatcherRarity["Uncommon"] = "Uncommon";
})(ParallelAssetMatcherRarity || (exports.ParallelAssetMatcherRarity = ParallelAssetMatcherRarity = {}));
/** The Parallel asset subtype. */
var ParallelAssetMatcherSubtype;
(function (ParallelAssetMatcherSubtype) {
    ParallelAssetMatcherSubtype["Clone"] = "Clone";
    ParallelAssetMatcherSubtype["Pirate"] = "Pirate";
    ParallelAssetMatcherSubtype["Vehicle"] = "Vehicle";
})(ParallelAssetMatcherSubtype || (exports.ParallelAssetMatcherSubtype = ParallelAssetMatcherSubtype = {}));
/** Attribute used to rank Parallel assets. */
var ParallelAssetRankingAttribute;
(function (ParallelAssetRankingAttribute) {
    ParallelAssetRankingAttribute["Attack"] = "attack";
    ParallelAssetRankingAttribute["Cost"] = "cost";
    ParallelAssetRankingAttribute["Health"] = "health";
    ParallelAssetRankingAttribute["LastPriceNetworkBaseToken"] = "lastPriceNetworkBaseToken";
    ParallelAssetRankingAttribute["LastPriceUsd"] = "lastPriceUsd";
    ParallelAssetRankingAttribute["Supply"] = "supply";
})(ParallelAssetRankingAttribute || (exports.ParallelAssetRankingAttribute = ParallelAssetRankingAttribute = {}));
var Plan;
(function (Plan) {
    Plan["Defined"] = "DEFINED";
    Plan["Enterprise"] = "ENTERPRISE";
    Plan["Free"] = "FREE";
    Plan["Standard"] = "STANDARD";
})(Plan || (exports.Plan = Plan = {}));
/** The type of NFT in the pool. */
var PoolNftType;
(function (PoolNftType) {
    PoolNftType["Erc721Erc20"] = "ERC721ERC20";
    PoolNftType["Erc721Eth"] = "ERC721ETH";
    PoolNftType["Erc1155Erc20"] = "ERC1155ERC20";
    PoolNftType["Erc1155Eth"] = "ERC1155ETH";
})(PoolNftType || (exports.PoolNftType = PoolNftType = {}));
/** Currency types for Prime pool events. */
var PrimePoolCurrency;
(function (PrimePoolCurrency) {
    PrimePoolCurrency["Eth"] = "ETH";
    PrimePoolCurrency["Prime"] = "PRIME";
})(PrimePoolCurrency || (exports.PrimePoolCurrency = PrimePoolCurrency = {}));
/** A Prime pool event type. */
var PrimePoolEventType;
(function (PrimePoolEventType) {
    PrimePoolEventType["Cache"] = "CACHE";
    PrimePoolEventType["CachingPaused"] = "CACHING_PAUSED";
    PrimePoolEventType["Claim"] = "CLAIM";
    PrimePoolEventType["EmergencyWithdraw"] = "EMERGENCY_WITHDRAW";
    PrimePoolEventType["EndTimestampUpdated"] = "END_TIMESTAMP_UPDATED";
    PrimePoolEventType["EthRewardsAdded"] = "ETH_REWARDS_ADDED";
    PrimePoolEventType["EthRewardsSet"] = "ETH_REWARDS_SET";
    PrimePoolEventType["LogPoolAddition"] = "LOG_POOL_ADDITION";
    PrimePoolEventType["LogPoolSetAllocPoint"] = "LOG_POOL_SET_ALLOC_POINT";
    PrimePoolEventType["LogSetPerSecond"] = "LOG_SET_PER_SECOND";
    PrimePoolEventType["LogUpdatePool"] = "LOG_UPDATE_POOL";
    PrimePoolEventType["PoolDiscovered"] = "POOL_DISCOVERED";
    PrimePoolEventType["RewardDecrease"] = "REWARD_DECREASE";
    PrimePoolEventType["RewardIncrease"] = "REWARD_INCREASE";
    PrimePoolEventType["TimeCachePeriodUpdated"] = "TIME_CACHE_PERIOD_UPDATED";
    PrimePoolEventType["Withdraw"] = "WITHDRAW";
})(PrimePoolEventType || (exports.PrimePoolEventType = PrimePoolEventType = {}));
/** The type of Prime pool caching contract. */
var PrimePoolType;
(function (PrimePoolType) {
    PrimePoolType["EthAndPrimeRewards"] = "ETH_AND_PRIME_REWARDS";
    PrimePoolType["PrimeRewards"] = "PRIME_REWARDS";
    PrimePoolType["TimedCacheEthAndPrimeRewards"] = "TIMED_CACHE_ETH_AND_PRIME_REWARDS";
})(PrimePoolType || (exports.PrimePoolType = PrimePoolType = {}));
var PublishingType;
(function (PublishingType) {
    PublishingType["Batch"] = "BATCH";
    PublishingType["Single"] = "SINGLE";
})(PublishingType || (exports.PublishingType = PublishingType = {}));
var QuoteCurrency;
(function (QuoteCurrency) {
    QuoteCurrency["Token"] = "TOKEN";
    QuoteCurrency["Usd"] = "USD";
})(QuoteCurrency || (exports.QuoteCurrency = QuoteCurrency = {}));
/** The quote token within the pair. */
var QuoteToken;
(function (QuoteToken) {
    QuoteToken["Token0"] = "token0";
    QuoteToken["Token1"] = "token1";
})(QuoteToken || (exports.QuoteToken = QuoteToken = {}));
var QuoteType;
(function (QuoteType) {
    QuoteType["Input"] = "INPUT";
    QuoteType["Output"] = "OUTPUT";
})(QuoteType || (exports.QuoteType = QuoteType = {}));
/** The order of ranking. */
var RankingDirection;
(function (RankingDirection) {
    RankingDirection["Asc"] = "ASC";
    RankingDirection["Desc"] = "DESC";
})(RankingDirection || (exports.RankingDirection = RankingDirection = {}));
/** Sandwich event label types. */
var SandwichLabelForEventType;
(function (SandwichLabelForEventType) {
    SandwichLabelForEventType["Backrun"] = "backrun";
    SandwichLabelForEventType["Frontrun"] = "frontrun";
    SandwichLabelForEventType["Sandwiched"] = "sandwiched";
})(SandwichLabelForEventType || (exports.SandwichLabelForEventType = SandwichLabelForEventType = {}));
var SimulateTokenContractBuySellErrorEnum;
(function (SimulateTokenContractBuySellErrorEnum) {
    SimulateTokenContractBuySellErrorEnum["InsufficientLiquidity"] = "INSUFFICIENT_LIQUIDITY";
    SimulateTokenContractBuySellErrorEnum["InsufficientOutputAmount"] = "INSUFFICIENT_OUTPUT_AMOUNT";
    SimulateTokenContractBuySellErrorEnum["TransferFailed"] = "TRANSFER_FAILED";
    SimulateTokenContractBuySellErrorEnum["UnknownError"] = "UNKNOWN_ERROR";
})(SimulateTokenContractBuySellErrorEnum || (exports.SimulateTokenContractBuySellErrorEnum = SimulateTokenContractBuySellErrorEnum = {}));
var SimulateTokenContractResultStatusEnum;
(function (SimulateTokenContractResultStatusEnum) {
    SimulateTokenContractResultStatusEnum["Failure"] = "FAILURE";
    SimulateTokenContractResultStatusEnum["Pending"] = "PENDING";
    SimulateTokenContractResultStatusEnum["Success"] = "SUCCESS";
})(SimulateTokenContractResultStatusEnum || (exports.SimulateTokenContractResultStatusEnum = SimulateTokenContractResultStatusEnum = {}));
var SparklineAttribute;
(function (SparklineAttribute) {
    SparklineAttribute["Price"] = "PRICE";
})(SparklineAttribute || (exports.SparklineAttribute = SparklineAttribute = {}));
/** Event types for a token. Mint or Burn. */
var TokenLifecycleEventType;
(function (TokenLifecycleEventType) {
    TokenLifecycleEventType["Burn"] = "BURN";
    TokenLifecycleEventType["Mint"] = "MINT";
})(TokenLifecycleEventType || (exports.TokenLifecycleEventType = TokenLifecycleEventType = {}));
/** The token of interest within a pair. Can be `token0` or `token1`. */
var TokenOfInterest;
(function (TokenOfInterest) {
    TokenOfInterest["Token0"] = "token0";
    TokenOfInterest["Token1"] = "token1";
})(TokenOfInterest || (exports.TokenOfInterest = TokenOfInterest = {}));
/** Token pair event types. */
var TokenPairEventType;
(function (TokenPairEventType) {
    TokenPairEventType["Burn"] = "BURN";
    TokenPairEventType["Buy"] = "BUY";
    TokenPairEventType["Collect"] = "COLLECT";
    TokenPairEventType["CollectProtocol"] = "COLLECT_PROTOCOL";
    TokenPairEventType["Mint"] = "MINT";
    TokenPairEventType["Sell"] = "SELL";
    TokenPairEventType["Swap"] = "SWAP";
    TokenPairEventType["Sync"] = "SYNC";
})(TokenPairEventType || (exports.TokenPairEventType = TokenPairEventType = {}));
/** The type of statistics returned. Can be `FILTERED` or `UNFILTERED` */
var TokenPairStatisticsType;
(function (TokenPairStatisticsType) {
    TokenPairStatisticsType["Filtered"] = "FILTERED";
    TokenPairStatisticsType["Unfiltered"] = "UNFILTERED";
})(TokenPairStatisticsType || (exports.TokenPairStatisticsType = TokenPairStatisticsType = {}));
/** The attribute used to rank tokens. */
var TokenRankingAttribute;
(function (TokenRankingAttribute) {
    /** @deprecated Use createdAt instead */
    TokenRankingAttribute["Age"] = "age";
    TokenRankingAttribute["BuyCount1"] = "buyCount1";
    TokenRankingAttribute["BuyCount4"] = "buyCount4";
    TokenRankingAttribute["BuyCount12"] = "buyCount12";
    TokenRankingAttribute["BuyCount24"] = "buyCount24";
    TokenRankingAttribute["Change1"] = "change1";
    TokenRankingAttribute["Change4"] = "change4";
    TokenRankingAttribute["Change12"] = "change12";
    TokenRankingAttribute["Change24"] = "change24";
    TokenRankingAttribute["CreatedAt"] = "createdAt";
    TokenRankingAttribute["High1"] = "high1";
    TokenRankingAttribute["High4"] = "high4";
    TokenRankingAttribute["High12"] = "high12";
    TokenRankingAttribute["High24"] = "high24";
    TokenRankingAttribute["Holders"] = "holders";
    TokenRankingAttribute["LastTransaction"] = "lastTransaction";
    TokenRankingAttribute["Liquidity"] = "liquidity";
    TokenRankingAttribute["Low1"] = "low1";
    TokenRankingAttribute["Low4"] = "low4";
    TokenRankingAttribute["Low12"] = "low12";
    TokenRankingAttribute["Low24"] = "low24";
    TokenRankingAttribute["MarketCap"] = "marketCap";
    TokenRankingAttribute["NotableHolderCount"] = "notableHolderCount";
    TokenRankingAttribute["PriceUsd"] = "priceUSD";
    TokenRankingAttribute["SellCount1"] = "sellCount1";
    TokenRankingAttribute["SellCount4"] = "sellCount4";
    TokenRankingAttribute["SellCount12"] = "sellCount12";
    TokenRankingAttribute["SellCount24"] = "sellCount24";
    TokenRankingAttribute["TrendingScore"] = "trendingScore";
    TokenRankingAttribute["TrendingScore1"] = "trendingScore1";
    TokenRankingAttribute["TrendingScore4"] = "trendingScore4";
    TokenRankingAttribute["TrendingScore12"] = "trendingScore12";
    TokenRankingAttribute["TrendingScore24"] = "trendingScore24";
    TokenRankingAttribute["TxnCount1"] = "txnCount1";
    TokenRankingAttribute["TxnCount4"] = "txnCount4";
    TokenRankingAttribute["TxnCount12"] = "txnCount12";
    TokenRankingAttribute["TxnCount24"] = "txnCount24";
    TokenRankingAttribute["UniqueBuys1"] = "uniqueBuys1";
    TokenRankingAttribute["UniqueBuys4"] = "uniqueBuys4";
    TokenRankingAttribute["UniqueBuys12"] = "uniqueBuys12";
    TokenRankingAttribute["UniqueBuys24"] = "uniqueBuys24";
    TokenRankingAttribute["UniqueSells1"] = "uniqueSells1";
    TokenRankingAttribute["UniqueSells4"] = "uniqueSells4";
    TokenRankingAttribute["UniqueSells12"] = "uniqueSells12";
    TokenRankingAttribute["UniqueSells24"] = "uniqueSells24";
    TokenRankingAttribute["UniqueTransactions1"] = "uniqueTransactions1";
    TokenRankingAttribute["UniqueTransactions4"] = "uniqueTransactions4";
    TokenRankingAttribute["UniqueTransactions12"] = "uniqueTransactions12";
    TokenRankingAttribute["UniqueTransactions24"] = "uniqueTransactions24";
    TokenRankingAttribute["Volume1"] = "volume1";
    TokenRankingAttribute["Volume4"] = "volume4";
    TokenRankingAttribute["Volume12"] = "volume12";
    TokenRankingAttribute["Volume24"] = "volume24";
    TokenRankingAttribute["VolumeChange1"] = "volumeChange1";
    TokenRankingAttribute["VolumeChange4"] = "volumeChange4";
    TokenRankingAttribute["VolumeChange12"] = "volumeChange12";
    TokenRankingAttribute["VolumeChange24"] = "volumeChange24";
})(TokenRankingAttribute || (exports.TokenRankingAttribute = TokenRankingAttribute = {}));
/** NFT marketplace names. */
var WebhookNftEventFillSource;
(function (WebhookNftEventFillSource) {
    WebhookNftEventFillSource["Blur"] = "BLUR";
    WebhookNftEventFillSource["Coinbase"] = "COINBASE";
    WebhookNftEventFillSource["Echelon"] = "ECHELON";
    WebhookNftEventFillSource["Element"] = "ELEMENT";
    WebhookNftEventFillSource["Ensvision"] = "ENSVISION";
    WebhookNftEventFillSource["Flipxyz"] = "FLIPXYZ";
    WebhookNftEventFillSource["Gem"] = "GEM";
    WebhookNftEventFillSource["Genie"] = "GENIE";
    WebhookNftEventFillSource["Kodex"] = "KODEX";
    WebhookNftEventFillSource["Magiceden"] = "MAGICEDEN";
    WebhookNftEventFillSource["Nftnerds"] = "NFTNERDS";
    WebhookNftEventFillSource["Opensea"] = "OPENSEA";
    WebhookNftEventFillSource["Rarible"] = "RARIBLE";
    WebhookNftEventFillSource["Reservoirtools"] = "RESERVOIRTOOLS";
    WebhookNftEventFillSource["Soundxyz"] = "SOUNDXYZ";
})(WebhookNftEventFillSource || (exports.WebhookNftEventFillSource = WebhookNftEventFillSource = {}));
/** NFT event types. */
var WebhookNftEventType;
(function (WebhookNftEventType) {
    WebhookNftEventType["Mint"] = "MINT";
    WebhookNftEventType["Sale"] = "SALE";
    WebhookNftEventType["Transfer"] = "TRANSFER";
})(WebhookNftEventType || (exports.WebhookNftEventType = WebhookNftEventType = {}));
/** The type of webhook. */
var WebhookType;
(function (WebhookType) {
    WebhookType["NftEvent"] = "NFT_EVENT";
    WebhookType["PriceEvent"] = "PRICE_EVENT";
    WebhookType["RawTransaction"] = "RAW_TRANSACTION";
    WebhookType["TokenPairEvent"] = "TOKEN_PAIR_EVENT";
})(WebhookType || (exports.WebhookType = WebhookType = {}));
var Join__Graph;
(function (Join__Graph) {
    Join__Graph["ApiAdmin"] = "API_ADMIN";
    Join__Graph["Decodings"] = "DECODINGS";
    Join__Graph["Meta"] = "META";
    Join__Graph["Nfts"] = "NFTS";
    Join__Graph["Tokens"] = "TOKENS";
    Join__Graph["Users"] = "USERS";
})(Join__Graph || (exports.Join__Graph = Join__Graph = {}));
var Link__Purpose;
(function (Link__Purpose) {
    /** `EXECUTION` features provide metadata necessary for operation execution. */
    Link__Purpose["Execution"] = "EXECUTION";
    /** `SECURITY` features provide metadata necessary to securely resolve fields. */
    Link__Purpose["Security"] = "SECURITY";
})(Link__Purpose || (exports.Link__Purpose = Link__Purpose = {}));
